lib/ansible/module_utils/tacp.py

```diff
@@ -347,3 +347,52 @@ def get_resource_by_uuid(uuid, component, api_client):
     #                 uuid=datacenter.uuid, uuid)
     #         except ApiException as e:
     #             return "Exception when calling get_firewall_profiles_using_get"
+
+
+class ApplicationGroupResource(object):
+    def __init__(self, api_client, name, datacenter_uuid=None):
+        self.api = tacp.ApplicationGroupsApi(api_client)
+        self.name = name
+        self.datacenter_uuid = datacenter_uuid
+
+    def get(self):
+        """ Returns the application group as a dict or None if nothing found.
+
+        eg: {'datacenter_uuid': '5b665e17-071d-4159-a0e0-58f4081ccd73',
+             'name': 'gulf',
+             'uuid': 'cef6c09c-d371-408e-8fd6-292befcc42d6'}
+
+        """
+        filters = f'name=={self.name}'
+        if self.datacenter_uuid:
+            filters += f';datacenterUuid=={self.datacenter_uuid}'
+
+        response = self.api.get_application_group_list_using_get(
+            filters=filters
+        )
+        if response:
+            return response[0].to_dict()
+        return None
+
+    def get_or_create(self):
```

Comment:
If you want to model this after get_or_create:
Djangos https://docs.djangoproject.com/en/3.0/ref/models/querysets/
They return the object first, then the created boolean. This would be cleaner.

Reply:
I didn't want to model anything, but that is where I've got the idea from. Modified.